src/query/api/v1/handler/prom/common.go

```diff
@@ -92,6 +92,8 @@ func Respond(w http.ResponseWriter, data interface{}, warnings promstorage.Warni
 
 // Responds with error status code and writes error JSON to response body.
 func RespondError(w http.ResponseWriter, err error) {
+	err = xhttp.GetErrorRewriteFn()(err)
```

Comment:
Hm, can't we just rewrite in `xhttp.WriteError`?

Reply:
Alas, `xhttp.WithErrorResponse(b)` passes already serialized error message, which is then directly written to HTTP response. Hence the rewrite inside `WriteError()` would only affect status code, the message would stay as in original error.

I was considering changing `xhttp.WithErrorResponse()` to accept a `func(error) []byte` instead of already prepared `[]byte`. What do you think about that?

Reply:
After #2933, both `RespondError()` and `WriteError()` emits `status` field. Thus, the latter can be used directly

https://github.com/m3db/m3/pull/3008/commits/6fb565d9c777d9475f21de8401b34f4da5b2a34f#diff-dcc137769b5c2ad03409d297cf05413b71396fc7df31a9a2ee6ecb9a9b20e22dL95-R95